### PR TITLE
Optimizations part N

### PR DESCRIPTION
**Pull request by roman-khimov**
Mostly allocation optimizations that allow us to more reliably cross the target TPS line. (needs a rebase after #2270 merge).

### CODE REVIEW

File: pkg/encoding/bigint/bigint.go [hunk 1]
```diff
@@ -15,6 +15,8 @@ const (
 	wordSizeBytes = bits.UintSize / 8
 )
 
+var bigOne = big.NewInt(1)
```
Comment (AnnaShaleva) [inline pkg/encoding/bigint/bigint.go:18]: We have the same (or `big.NewInt(0)`) var declarations in several places (i.g. for native contracts, stackitems and VM code), so maybe create a separate helper package for them?

Comment (roman-khimov) [reply pkg/encoding/bigint/bigint.go:18]: Not all of them can legitimately be reused, in some cases we do need a separate integer every time we use it. But if they can, then `bigint` probably is the right package for them. I'll check which ones can be used this way.

Comment (roman-khimov) [reply pkg/encoding/bigint/bigint.go:18]: After looking at it for while I'd say that private constants still make a bit more sense:
 * less chances things will go wild if something changes the variable
 * packages have only things they need (no useless constants or functionality from `bigint`)

Sure, `big.NewInt(1)` is a widely used thing, but 3 or 100 not so much. There will be some duplicates, but they don't cost a lot.

File: pkg/vm/context.go [hunk 2]
```diff
@@ -277,16 +277,19 @@ func (c *Context) DumpStaticSlot() string {
...
 // DumpArgumentsSlot returns json formatted representation of the given slot.
 func (c *Context) DumpArgumentsSlot() string {
-	return dumpSlot(c.arguments)
+	return dumpSlot(&c.arguments)
 }
 
 // dumpSlot returns json formatted representation of the given slot.
-func dumpSlot(s *Slot) string {
+func dumpSlot(s *slot) string {
+	if s == nil {
```
[the remark this answers is not in the document]
Comment (roman-khimov) [reply pkg/vm/context.go]: Fixed.

**Review by fyrchik — APPROVED** [review]

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**